lib/rucio/core/rule.py

```diff
@@ -750,10 +754,11 @@ def list_rules(filters={}, *, session: "Session"):
             query = query.filter(getattr(models.ReplicationRule, key) == value)
 
     try:
-        for rule in query.yield_per(5):
+        for rule, data_identifier_bytes in query.yield_per(5):
             d = {}
             for column in rule.__table__.columns:
                 d[column.name] = getattr(rule, column.name)
```

Comment:
Please take the opportunity to replace with `to_dict()` (in a separate commit).

Reply:
So I need to make another commit to this pr and then just squash all the commits?

Reply:
I always recommend changes that are supposed to maintain exact functionality to be isolated and merged as separate commits (and not squashed). This aids the review process, but also future debugging. Consequently, I find it reasonable for this PR to be made out of three commits: the SQLAlchemy 2.0 migration, this replacement with `to_dict()`, and finally the changes that close the issue related to this PR.

Reply:
You’ll need to take some time to rewrite your commit history in order to achieve what I described above.

Reply:
Sure, no problem

Reply:
Despite the latest changes, the commit history has room for improvement.

 1. The subject of the first commit is `Rules: SQLAlchemy 2.0 migration #5978`, yet it modifies `bin/rucio` and changes the structure return by `list_rules()`. Also, it introduces some code formatted unconventionally, that is then improved by the third commit.
 2. The second commit is perfect.
 3. The third commit will have to be adjusted, once the first commit is fixed.

In short, the end result (meaning the three commits combined) is perfect. However, your individual commits are introducing some mistakes (e.g. `size` instead of `bytes`) and fixing them afterwards whereas you should be introducing the final iteration of your code immediately.

Let me know in case I’m not being clear or you require assistance.

Reply:
I rewrote commit history. Please, have a look, it seems to be complete: I deleted and edited changes in my local branch to the point it had only SQLAlchemy 2.0 migration, pushed it, added `to_dict()` - pushed another commit and then made final commit. In total I had 6 commits in PR and squashed 3 of them. Such operations don't seem to be very effective. Could I have done it easier and faster?

Reply:
Yes, this is exactly what I was hoping for. Thanks!

`git rebase --interactive` allows all sort of rewrites of your commit history (e.g. edit, merge, split, reorder, and drop commits) in a systematic way. In this particular case, I would have expected something along the lines of:

 1. Start the rebase at the base of your branch, `edit` the first and third commit and `pick` the second commit.
 2. Edit the two files as appropriately, then amend the commit. Alternatively, undo the commit with `git reset`, edit, then commit anew.
 3. Continue the rebase. The second commit should be applied automatically.
 4. The third commit will pose a conflict, due to the changes applied on the first commit. Simply edit (and reintroduce the changes to `/bin/rucio` which were removed in the first commit) until you’re happy with the end result.
 5. Finish the rebase.

It might be worth looking up for some specific examples first, but once gotten used to, it will become second nature.

Feel free to mark this conversation as resolved after you’re read my response.

Reply:
Such approach seems to be more convenient. Thanks! I'll keep that in mind.